smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/ServiceAggregatedClientGenerator.java

```diff
@@ -83,43 +92,37 @@ public void run() {
                 );
 
                 // Generate a multiple overloaded methods for each command.
-                writer.writeShapeDocs(operation);
-                writer.write("public $L(\n"
+                writer.writeDocs(
+                    "@see {@link " + operationSymbol.getName() + "}"
+                );
+                writer.write("$L(\n"
                             + "  args: $T,\n"
                             + "  options?: $T,\n"
                             + "): Promise<$T>;", methodName, input, applicationProtocol.getOptionsType(), output);
-                writer.write("public $L(\n"
+                writer.write("$L(\n"
                              + "  args: $T,\n"
                              + "  cb: (err: any, data?: $T) => void\n"
                              + "): void;", methodName, input, output);
-                writer.write("public $L(\n"
+                writer.write("$L(\n"
                             + "  args: $T,\n"
                             + "  options: $T,\n"
                             + "  cb: (err: any, data?: $T) => void\n"
                             + "): void;", methodName, input, applicationProtocol.getOptionsType(), output);
-                writer.openBlock("public $1L(\n"
-                                 + "  args: $2T,\n"
-                                 + "  optionsOrCb?: $3T | ((err: any, data?: $4T) => void),\n"
-                                 + "  cb?: (err: any, data?: $4T) => void\n"
-                                 + "): Promise<$4T> | void { ", "}",
-                        methodName,
-                        input,
-                        applicationProtocol.getOptionsType(),
-                        output,
-                        () -> {
-                    writer.write("const command = new $T(args);\n"
-                                 + "if (typeof optionsOrCb === \"function\") {\n"
-                                 + "  this.send(command, optionsOrCb)\n"
-                                 + "} else if (typeof cb === \"function\") {\n"
-                                 + "  if (typeof optionsOrCb !== \"object\")\n"
-                                 + "    throw new Error(`Expect http options but get $${typeof optionsOrCb}`)\n"
-                                 + "  this.send(command, optionsOrCb || {}, cb)\n"
-                                 + "} else {\n"
-                                 + "  return this.send(command, optionsOrCb);\n"
-                                 + "}", operationSymbol);
-                });
                 writer.write("");
             }
         });
+
+        writer.write("");
+
+        writer.addImport(barebonesClientName + "Config", null, serviceSymbol.getNamespace());
```

Comment:
Maybe we should add another method in ServiceBareBonesClientGenerator for getting the config type name, like https://github.com/awslabs/smithy-typescript/blob/a526a73f8fada65c9ef0bddea94a85695229f2c4/smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/ServiceBareBonesClientGenerator.java#L90